File: resource-registry/src/test/java/nl/haploid/octowight/registry/repository/ResourceRootDmoFactoryTest.java
package nl.haploid.octowight.registry.repository;

import mockit.Tested;
import nl.haploid.octowight.registry.TestData;
import nl.haploid.octowight.registry.data.ResourceRoot;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class ResourceRootDmoFactoryTest {

	@Tested
	private ResourceDmoFactory resourceDmoFactory;

	@Test
	public void testFromResource() {
		final ResourceRoot resourceRoot = TestData.resourceRoot(123l);
		final ResourceDmo resourceDmo = resourceDmoFactory.fromResource(resourceRoot);
		assertEquals(resourceRoot.getAtomId(), resourceDmo.getAtomId());
		assertEquals(resourceRoot.getAtomType(), resourceDmo.getAtomType());
		assertEquals(resourceRoot.getAtomLocus(), resourceDmo.getAtomLocus());
		assertEquals(resourceRoot.getResourceId(), resourceDmo.getResourceId());
		assertEquals(resourceRoot.getResourceType(), resourceDmo.getResourceType());
	}
}

File: resource-registry/src/test/scala/nl/haploid/octowight/registry/repository/ResourceRootDmoFactoryTest.scala
package nl.haploid.octowight.registry.repository

import nl.haploid.octowight.registry.{AbstractTest, TestData}

class ResourceRootDmoFactoryTest extends AbstractTest {

  "Resource root DMO factory" should "build DMO from resource root" in {
    val resourceRoot = TestData.resourceRoot(123l)
    val resourceRootDmo = ResourceRootDmo(resourceRoot)
    resourceRootDmo.getRoot.getId should be(resourceRoot.getRoot.id)
    resourceRootDmo.getRoot.getCategory should be(resourceRoot.getRoot.category)
    resourceRootDmo.getRoot.getOrigin should be(resourceRoot.getRoot.origin)
    resourceRootDmo.getResourceId should be(resourceRoot.getResourceId)
    resourceRootDmo.getResourceType should be(resourceRoot.getResourceType)
  }
}
